draco/core/misc.py

```diff
@@ -214,6 +217,36 @@ def finish(self):
         return items
 
 
+class CheckMPIEnvironment(task.MPILoggedTask):
+    """Check that the current MPI environment can communicate
+    across all nodes.
+    """
+
+    timeout = config.Property(proptype=int, default=60)
+
+    def setup(self):
+        from caput import mpiutil
+        import time
+
+        comm = mpiutil._comm
+        results = []
+
+        # Send and receive across all ranks
+        for i in range(comm.size):
+            results.append(comm.Irecv(np.zeros(1), (comm.rank - i) % comm.size))
+            comm.Isend(np.ones(1), (comm.rank + i) % comm.size)
+
+        time.sleep(self.timeout)
```

Comment:
I might send something larger in here to stress it a bit more. Maybe a few MB per message.

Also, you might want to put a testable pattern in the message to ensure it's the correct, uncorrupted message, e.g. alternating elements of the sending and receiving rank.